Bottom: 8acc4f5f1cd0cc3ae2ffea5ff8583c029904c895
Top:    a92d86e39eaef4298810db13529db753b7a9cf46
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:55 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 511bd21..0000871 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2815,12 +2815,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
